### PR TITLE
More quick equip compatiability and fixes

### LINKED ISSUE #16992: Recent quick equip changes cause a few issues.
With the recent quick equip changes there has been a few issues I've come across.

- Trying to quick equip a small item to your pockets will try to swap the item in pocket(left) before it puts it into the second pocket(right). Having item in the right one and then trying it will properly insert the item into the left side but after that the above issues happens.
- Belt items like emergency air tanks will try to go to the belt first but it will swap the belt item even if you have an open pocket or suit armor slot. I had a hardsuit on and tried it and it would swap the E airtank to the belt first even when I had something in the belt already. It should have gone to the open armor/suit slot or the pocket before that.
The old behavior would try to put it in the suit slot first and then try to put it in pockets and then on the belt.

### LINKED ISSUE #34406: Screwdriver quickswap defaults to wallet over toolbelts
(WEB REPORT BY: loppu REMOTE: 206.221.180.138:7777)
# Revision
e7e8bc77248c37512b0d154cdfe1365bb502c551

# Description
Pressing equip hotkey 'e' while a screwdriver is in your hands will prioritize sending it into your wallet instead of your toolbelt. This is annoying.

# Steps to Reproduce
Place screwdriver in hand with toolbelt and wallet equipped. Press e.

# What you Expected
Screwdriver goes where it should.

# What Actually Happened
Very minor irritation and reduced qol.

### LINKED ISSUE #34762: Screwdrivers Prefer Wallets Over Toolbelts
(WEB REPORT BY: dovemite REMOTE: 206.221.180.138:7777)
# Revision
c1aece31bbdc1dc329e82a3576acdaa5732ff8f6

# Description
When quickly swapping between tools, screwdrivers will go into wallets rather than the toolbelt. This is quite annoying when the rest of your tools are in your belt.
# Steps to Reproduce
Quickly store away a screwdriver while you have a wallet
# What you Expected
Screwdriver to go into my toolbelt rather with my other tools
# What Actually Happened
Screwdriver goes into wallet and I have to go pull it back out everytime

### PR DESCRIPTION
**Pull request by SECBATON-GRIFFON**
[tweak]

## What this does
Extends functionality to attaching accessories and holstering, adds a priority system for certain items to quick equip to certain slots first.
Closes #34406.
Closes #34762.
Handles the screwdriver and cigarette ID equip issues by whitelisting it to only stuff that really needs to be in that slot for priority.
One idea I also had in mind was using quick equip with an empty hand to unholster from any available holster, but that seems too contentious for this PR to be merged with everything else.

## Why it's good
Makes accessories and holstering faster to use.

## Changelog
:cl:
 * tweak: Clothing accessories can now be attached with quick equip
 * tweak: Items can now be holstered with quick equip. Currently, only handguns prioritise this.
 * tweak: Only wallets, cartridges, pens, IDs, PDAs, cash and coins will now go into the ID slot before anything else.

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -648,7 +648,8 @@ Use this proc preferably at the end of an equipment loadout
 		return 0
 
 	var/list/backup_slots = list()
-	for(var/slot in slot_equipment_priority)
+	var/list/slots_to_use = W.quick_equip_priority | slot_equipment_priority
```
Comment (jwhitak) [inline code/modules/mob/mob.dm:651]: This is actually a fun trick to reorder a list!

**Comment by Dacendeth** [comment]
Tested this, everything seems to work as it should.

**Review by jwhitak — APPROVED** [review]
I had our boy Razzle Dazzle run some testing on this code and everything's working as intended. No immediately notable game breaking problems! Quick holstering, rapid accessorizing, and no more screwdrivers in the wallet... I'm not sure if this is a mere bug fix, but it works!

**Comment by hacker-on-steroids** [comment]
speaking of quick equip, i was thinking about making it prioritise putting items into empty slots, then backpack/belt, and if it can't do that, swap with existing items
so you could quickstore weapons onto armor vests, or quickstore a wrench into your backpack instead of swapping your medbelt

wanna know if it's something people would want, or just a skill issue on my end

**Comment by Dacendeth** [comment]
@hacker-on-steroids God I wish it always prioritized empty slots, I absolutely hate quick equipping a health analyzer while wearing a toolbelt with an empty pocket
My ideal setup would be something like... If an item would swap it first check for an empty pocket if it fits there and go in if it does, then check for empty suit storage and go in there if it does, then finally check for backpack slots and go in there if it does and if after all that it cant fit anywhere it swaps with something.

**Pull request merged**